### PR TITLE
Cursor Plugin: Flip time text to left of cursor on right edge for readability

### PR DESCRIPTION
**Pull request by mdtanrikulu**
Hey there!

I did a small change on cursor plugin, which improves the readability of duration text next to the cursor. When user point the very right edge, the duration will be placed on left side of the cursor for better UX. See the gif; 

**Before:**
duration_without_flip

**After:**
duration_flip

### CODE REVIEW

**Comment by thijstriemstra** [comment]
awesome, thanks!

File: src/plugin/cursor.js [hunk 1]
```diff
@@ -311,4 +321,20 @@ export default class CursorPlugin {
...
         );
     }
+
+    /**
+     * Get outer width of given element.
+     *
+     * @param {DOM} e DOM Element
+     * @returns {number} outer width
+     */
+    outerWidth(e) {
```
Comment (thijstriemstra) [inline src/plugin/cursor.js]: can you use a more descriptive var name then `e`?

File: src/plugin/cursor.js [hunk 2]
```diff
@@ -311,4 +321,20 @@ export default class CursorPlugin {
...
+        if (!e) return 0;
+
+        var width = e.offsetWidth;
```
Comment (thijstriemstra) [inline src/plugin/cursor.js]: use let instead of var

File: src/plugin/cursor.js [hunk 3]
```diff
@@ -200,12 +201,15 @@ export default class CursorPlugin {
                         {
                             display: 'inline',
                             pointerEvents: 'none',
-                            margin: 'auto'
+                            margin: 'auto',
+                            visibility: 'hidden' //initial value will be hidden just for measuring purpose
```
Comment (thijstriemstra) [inline src/plugin/cursor.js]: space between // and initial

File: src/plugin/cursor.js [hunk 4]
```diff
@@ -200,12 +201,15 @@ export default class CursorPlugin {
...
                         },
                         this.params.customShowTimeStyle
                     )
                 )
             );
+            //initial value to measure display width
```
Comment (thijstriemstra) [inline src/plugin/cursor.js]: space between // and initial

File: src/plugin/cursor.js [hunk 5]
```diff
@@ -200,12 +201,15 @@ export default class CursorPlugin {
...
+            this.displayTime.innerHTML = '0:00:000';
```
Comment (thijstriemstra) [inline src/plugin/cursor.js]: instead of hardcoding the time value, use the (user-supplied or default) format time method instead.

Comment (mdtanrikulu) [reply src/plugin/cursor.js]: Oh, I didn't see there is custom format method. Thanks for pointing it, I've updated.

File: src/plugin/cursor.js [hunk 6]
```diff
@@ -255,9 +259,14 @@ export default class CursorPlugin {
             const timeValue =
                 Math.max(0, (xpos / elementWidth) * duration) + scrollTime;
             const formatValue = this.formatTime(timeValue);
+            if (flip) {
+                const textOffset = this.outerWidth(this.displayTime);
+                xpos -= textOffset;
+            }
```
Comment (aburai) [inline src/plugin/cursor.js]: can be shorten to `if (flip) xpos -= this.outerWidth(this.displayTime);`

File: src/plugin/cursor.js [hunk 7]
```diff
@@ -90,13 +90,14 @@ export default class CursorPlugin {
         const bbox = this.wavesurfer.container.getBoundingClientRect();
         let y = 0;
         let x = e.clientX - bbox.left;
+        let flip = bbox.right < e.clientX + this.outerWidth(this.displayTime);
```
Comment (aburai) [inline src/plugin/cursor.js]: we should use `const` where possible

**Comment by aburai** [comment]
ups, already merged, i oversee that ;)

**Pull request merged**